doc/map.md

```diff
@@ -0,0 +1,541 @@
+        
+Use them to figure out which purpose each of the item types in the `datafile.item_types` section of the datafile has.
+
+Things to keep in mind:
+1. When an item type appears in `datafile.item_types`, it means that there must be at least one item of that type
+2. With the exception fo the UUID Index, the first item of an item type will have `id` = 0 and from there it will count up
+
+UUID item types
+---------------
+
+In DDNet, some item types won't be assigned a type_id, but instead an uuid.
+
+To find the correct item type (in `datafile.item_types` for uuid item types, you will need their `type_id`.
+You will need to figure out the `type_id` manually by looking into the **UUID Index items**.
+
+    UUID Index Item structure:
+        type_id: 0xffff
+        id: type_id of the uuid item type that this item represents
+        item_data:
+            [3] UUID of the uuid item type that this item represents
+
+- the twelve bytes of the uuid are laid out in order in the `item_data`.
+- steps to find an uuid item type:
+    1. get the UUID item type
+    2. scan through its items
+    3. when an item has the correct uuid in its `item_data`, copy the `type_id` from the `id` field
+    4. find the item type with the `type_id` that we just found out
+
+Map Item Types
+==============
+
+Version
+---------
+
+- `type_id` = 0
+- exactly one item
+
+
+    item_data of the only version item:
+        [1] version
+
+- `version` = 1
+- no actual data is stored using the Version item type
+
+Info
+------
+
+- `type_id` = 1
+- exactly one item
+
+
+    item_data of the only version item:
+        [1] (item) version
+        [1] opt &author: CString
+        [1] opt &version: CString
+        [1] opt &credits: CString
+        [1] opt &license: CString
+        [1] opt &settings: [CString] (DDNet only)
+
+- both vanilla and DDNet are at `version` = 1
+- like indicated, all the other fields are optional data item indices
+- the data item behind `settings` is an array of CStrings, all consecutive, split by their null bytes (with a null byte at the very end)
+- maximum amount of bytes for each CString (including the null byte):
+    - author: 32
+    - version: 16
+    - credits: 128
+    - license: 32
+
+Images
+------
+
+- `type_id` = 2
+
+
+    item_data of image items:
+        [1] version
+        [1] width
+        [1] height
+        [1] external: bool
+        [1] &name: CString
+        [1] opt &data: [Pixel]
+
+        version 2 extension (Vanilla only):
+        [1] variant
+
+- Vanilla is at `version` = 2, DDNet is at `version` = 1
+- `width` and `height` specify the dimensions of the image
+- if `version` = 1, the image is of type RGBA, for `version` = 2 `variant` holds the type:
+    - 0 -> RGB
+    - 1 -> RGBA
+- Images can either be embedded or external.
+    - Embedded images have `external` = false and have the image data stored in the data field.
+    The image data is simply a 2d-array of pixels.
```

Comment:
I had to look up the words: https://en.wikipedia.org/wiki/Row-_and_column-major_order

Are the pixels in row- or column-major order?